roles/roles_server.js

```diff
@@ -1,25 +1,19 @@
-/* global Meteor, Roles */
-let indexFnAssignment
-let indexFnRoles
+import { Meteor } from 'meteor/meteor'
+import { RolesCollection, RoleAssignmentCollection } from './roles_common_async'
 
-if (Meteor.roles.createIndexAsync) {
-  indexFnAssignment = Meteor.roleAssignment.createIndexAsync.bind(Meteor.roleAssignment)
-  indexFnRoles = Meteor.roles.createIndexAsync.bind(Meteor.roles)
-} else if (Meteor.roles.createIndex) {
-  indexFnAssignment = Meteor.roleAssignment.createIndex.bind(Meteor.roleAssignment)
-  indexFnRoles = Meteor.roles.createIndex.bind(Meteor.roles)
-} else {
-  indexFnAssignment = Meteor.roleAssignment._ensureIndex.bind(Meteor.roleAssignment)
-  indexFnRoles = Meteor.roles._ensureIndex.bind(Meteor.roles)
-}
+const indexFnAssignment = RoleAssignmentCollection.createIndexAsync.bind(RoleAssignmentCollection)
+const indexFnRoles = RolesCollection.createIndexAsync.bind(RolesCollection)
 
-[
+const indexes = [
   { 'user._id': 1, 'inheritedRoles._id': 1, scope: 1 },
   { 'user._id': 1, 'role._id': 1, scope: 1 },
   { 'role._id': 1 },
   { scope: 1, 'user._id': 1, 'inheritedRoles._id': 1 }, // Adding userId and roleId might speed up other queries depending on the first index
   { 'inheritedRoles._id': 1 }
-].forEach(index => indexFnAssignment(index))
+]
+for (const index of indexes) {
+  indexFnAssignment(index)
```

Comment:
shouldn't there be an `await` here?

Reply:
No. Creation of indexes is not a priority that needs to hold up the application.